modin/engines/base/frame/data.py

```diff
@@ -1702,31 +1708,48 @@ def _copartition(self, axis, other, how, sort, force_repartition=False):
         right_old_idxes = index_other_obj
 
         is_avoid_reindex = len(joined_index) != len(joined_index.unique()) and axis == 0
```

Comment:
`is_avoid_reindex ` is used only in one place in current state. Can we use boolean expression directly in [1713](https://github.com/modin-project/modin/pull/2367/files#diff-64aee86078c1a929c18582ab58f1bc06e251bf21fbf10f523ce433f56ab7b8b0R1713)?

Reply:
Yes, we can. Fixed.